store/dataStore.ts

```diff
@@ -1,17 +1,33 @@
 import { create } from 'zustand';
+import { produce } from 'immer';
```

Comment:
M: need to add a unit tests for this file.

Reply:
✅ Unit test added.